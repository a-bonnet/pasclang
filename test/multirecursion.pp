program
    var n : integer;

function fib(n : integer) : integer;
begin
    if n = 0 then
        fib := 0
    else if n = 1 then
        fib := 1
    else
        fib := fib1(n) + fib2(n)
end;

function fib1(n : integer) : integer;
begin
    n := n - 1;
    if n = 0 then
        fib1 := 0
    else if n = 1 then
        fib1 := 1
    else fib1 := fib1(n) + fib2(n)
end;

function fib2(n : integer) : integer;
begin
    n := n - 2;
    if n = 0 then
        fib2 := 0
    else if n = 1 then
        fib2 := 1
    else fib2 := fib1(n) + fib2(n)
end;

begin
    n := 1;
    while n <> 0 do
        begin
            n := readln();
            writeln(fib(n))
        end
end.

