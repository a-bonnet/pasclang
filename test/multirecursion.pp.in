3
2
1
8
15
5
0
