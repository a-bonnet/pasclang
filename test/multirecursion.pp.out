2
1
1
21
610
5
0
